content/en/blog/_posts/2023-01-12-protect-mission-critical-pods-priorityclass/index.md

```diff
@@ -0,0 +1,300 @@
+---
+layout: blog
+title: "Protect Your Mission-Critical Pods From Eviction With PriorityClass"
+date: 2023-01-12
+slug: protect-mission-critical-pods-priorityclass
+description: "Pod priority and preemption help to make sure that mission-critical pods are up in the event of a resource crunch by deciding order of scheduling and eviction."
+---
+
+
+**Author:** Sunny Bhambhani (InfraCloud Technologies)
+
+Kubernetes has been widely adopted, and many organizations use it as their de-facto orchestration engine for running workloads that need to be created and deleted frequently.
+
+Therefore, proper scheduling of the pods is key to ensuring that application pods are up and running within the Kubernetes cluster without any issues. This article delves into the use cases around resource management by leveraging the [PriorityClass](/docs/concepts/scheduling-eviction/pod-priority-preemption/#priorityclass) object to protect mission-critical or high-priority pods from getting evicted and making sure that the application pods are up, running, and serving traffic.
+
+## Resource management in Kubernetes 
+
+The control plane consists of multiple components, out of which the scheduler (usually the built-in [kube-scheduler](/docs/concepts/scheduling-eviction/kube-scheduler/)) is one of the components which is responsible for assigning a node to a pod.
+- Whenever a pod is created, it goes to a pending state, and then the scheduler kicks in and determines which node should be the best-suited node for the placement of the new pod.
```

Comment:
Its just about listing the scheduling steps, do let me know if you want me to use paragraph intead of bullet points.

Reply:
A paragraph would be better because some of these steps are happening in parallel/in the background. A list, in my opinion, is for events occurring in a particular order/sequence.

Reply:
Updated:

`
In short, whenever a pod is created, it enters a "pending" state, after which the scheduler determines which node should be the best-suited node for the placement of the new pod.

In the background, the scheduler runs as an infinite loop looking for pods without a `nodeName` set and that are [ready for scheduling](/docs/concepts/scheduling-eviction/pod-scheduling-readiness/). For each Pod that needs scheduling, the scheduler tries to make its best decision about which node should run that Pod.

Just in case the scheduler is not able to find any node, the pod remains in the pending state, which is not an ideal scenario.

{{< note >}}
To name a few, `nodeSelector` , `taints and tolerations` , `nodeAffinity` , the rank of nodes based on available resources (for example, CPU and memory), and several other criteria are used to determine the pod's placement.
{{< note >}}
`

Reply:
Whenever a pod is created, it enters a "pending" state, after which the scheduler determines which node is best suited for the placement of the new pod.

In the background, the scheduler runs as an infinite loop looking for pods without a nodeName set that are [ready for scheduling](https://github.com/docs/concepts/scheduling-eviction/pod-scheduling-readiness/). For each Pod that needs scheduling, the scheduler tries to decide which node should run that Pod.

If the scheduler cannot find any node, the pod remains in the pending state, which is not ideal.

{{< note >}}
To name a few, nodeSelector , taints and tolerations , nodeAffinity , the rank of nodes based on available resources (for example, CPU and memory), and several other criteria are used to determine the pod's placement.
{{< note >}}

Reply:
Also, the figure below is getting rendered as a note.

Reply:
Corrected, missed to close note.